### PR TITLE
Correct property EBNF for unquoted strings

### PR DESCRIPTION
**Pull request by paulidale**
Replacing #19998.

This is the minimal change.  It would be prudent for unquoted strings to be less liberal in what is allowed to permit future expansions.  The question then becomes what?

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Comment by slontis** [comment]
Is it possible to add a test?

**Comment by paulidale** [comment]
Testing what exactly?  There are tests for unquoted strings already and this is in no way changing the behaviour of the code -- it's purely the grammar.

**Comment by levitte** [comment]
Of course you can add a test or two!  #19998 did exactly that.  Negative tests (tests meant to fail) wouldn't be a bad idea, for example.

File: crypto/property/properties.ebnf [hunk 1]
```diff
@@ -10,7 +10,7 @@ Value    ::= NumberLiteral
 StringLiteral ::= QuotedString | UnquotedString
 QuotedString     ::= '"' [^"]* '"'
         | "'" [^']* "'"
-UnquotedString ::= [^{space},]+
+UnquotedString ::= [A-Z] [^{space},]+
```
Comment (levitte) [inline crypto/property/properties.ebnf:13]: Only upper case?  I ask, because NumberLiteral specifies A-F *and* a-f...

File: doc/man7/property.pod [hunk 2]
```diff
@@ -154,7 +154,7 @@ The lexical syntax in EBNF is given by:
  Value          ::= NumberLiteral | StringLiteral
  StringLiteral  ::= QuotedString | UnquotedString
  QuotedString   ::= '"' [^"]* '"' | "'" [^']* "'"
- UnquotedString ::= [^{space},]+
+ UnquotedString ::= [A-Z] [^{space},]+
```
Comment (levitte) [inline doc/man7/property.pod:157]: Same thing here

**Review by levitte — COMMENTED** [review]
... one might wonder about PropertyName too...

**Comment by levitte** [comment]
Having tried to use this spec, and having a re-read of them now, I find them increasingly hard to follow.  For example, the way they are written, it looks like only upper case characters are allowed for property names, 'cause the number literal explicitly specifies both `A-F` and `a-f`, indicating to me that letters are case sensitive (which they are in most EBNFs, but not in ABNF).

So we end up with a spec that leaves room for trial and error, or has anyone who wants to know the *actual* rules to find the source and read that instead.  That's exactly what I had to do when a `x.author=@levitte` in my provider sent it tumbling into a chaos of errors.

**Review by levitte — CHANGES_REQUESTED** [review]
Please add a reference to the EBNF grammar in `doc/man7/property.pod`.  `crypto/property/properties.ebnf` mentions https://bottlecaps.de/rr/ui, that should be in the public docs.

**Review by levitte — CHANGES_REQUESTED** [review]
Please change all `A-F` to `A-Fa-f` where that's not already the case.  The EBNF grammar that you have used considers letter case sensitive as far as I can tell.

**Comment by levitte** [comment]
WIth those two changes, I'll be satisfied enough

**Comment by paulidale** [comment]
There is no benefit adding a link to the grammar in crypto/property.  The two grammars are completely identical.  Instead I've removed the duplication and left the documentation as the canonical reference.

**Comment by paulidale** [comment]
Negative test added.

**Comment by paulidale** [comment]
Ping for review

**Review by slontis — APPROVED** [review]
Should this also have a ref to the EBNF standard being used.. (There seem to be multiple of them).

**Comment by levitte** [comment]
> Should this also have a ref to the EBNF standard being used.. (There seem to be multiple of them).

Yup. That was one of my change requests:

https://github.com/openssl/openssl/pull/20023#pullrequestreview-1243695382

**Comment by paulidale** [comment]
Link to definition of EBNF added FWIW.

**Review by hlandau — APPROVED** [review]

**Review by levitte — APPROVED** [review]

**Comment by paulidale** [comment]
Merged.
